javascriptlib/src/mill/javascriptlib/PublishModule.scala

```diff
@@ -0,0 +1,539 @@
+package mill.javascriptlib
+
+import mill.*
+import os.*
+
+trait PublishModule extends TypeScriptModule {
+
+  /**
+   * Metadata about your project, required to publish.
+   *
+   * This is an equivalent of a `package.json`
+   */
+  def publishMeta: T[PublishModule.PublishMeta]
+
+  override def npmDevDeps: T[Seq[String]] =
+    Task { Seq("glob@^10.4.5", "ts-patch@3.3.0", "typescript-transform-paths@3.5.3") }
+
+  def bundledOut: T[String] = Task { "dist" }
+
+  private def declarationOut: T[String] = Task { "declarations" }
+
+  override def mainFileName: T[String] = Task { s"${millSourcePath.last}.js" }
+
+  // main file; defined with mainFileName
+  def main: T[String] =
+    Task { bundledOut() + "/src/" + mainFileName() }
+
+  private def mainType: T[String] = Task {
+    main().replaceFirst(bundledOut(), declarationOut()).replaceAll("\\.js", ".d.ts")
+  }
+
+  // Define exports for the package
+  // by default: mainFile is exported
+  // use this to define other exports
+  def exports: T[Map[String, String]] = Task { Map.empty[String, String] }
+
+  private def mkExports: T[Map[String, PublishModule.ExportEntry]] = Task {
```

Comment:
All these task names should be prefixed with `publish` so it's clear what they are for and what trait they came from